src/Helpers/ProjectHelpers.cs

```diff
@@ -132,9 +133,21 @@ public static ProjectItem AddFileToProject(this Project project, FileInfo file,
 
 			// Appears to be a bug in project system or VS, so need to make sure the project is aware of the folder structure first,
 			// then add the time to that folderStructure (create and/or find)
-			ProjectItem folderItem = project.ProjectItems.AddFolder(file.DirectoryName);
-			ProjectItem item = folderItem.ProjectItems.AddFromTemplate(file.FullName, file.Name);
+			// if adding to the root ProjectItem then just do that.
+			ProjectItems projectItems;
+			if (string.Equals(root.TrimEnd(Path.DirectorySeparatorChar), file.DirectoryName, StringComparison.InvariantCultureIgnoreCase))
```

Comment:
Looks like it would do the trick for single sub folders, but what if a user typed `folder1/folder2/folder3/foo.cs`?

Reply:
Works (see line 144)
![nested](https://user-images.githubusercontent.com/4821/146420282-13f60efa-75c9-433a-ad1e-6f8cb994ee5d.gif)

Reply:
OMG, that's amazing!!! Thank you, thank you, thank you!!